trendmicro_deepsecurity/icon_trendmicro_deepsecurity/actions/deploy_rules/action.py

```diff
@@ -0,0 +1,77 @@
+import komand
+from .schema import DeployRulesInput, DeployRulesOutput, Input, Output, Component
+# Custom imports below
+
+import requests
+import json
+
+from icon_trendmicro_deepsecurity.util.shared import tryJSON
+from icon_trendmicro_deepsecurity.util.shared import checkResponse
+
+class DeployRules(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='deploy_rules',
+                description=Component.DESCRIPTION,
+                input=DeployRulesInput(),
+                output=DeployRulesOutput())
+
+    def run(self, params={}):
+        """
+        Set IPS rules in Deep Security
+        """
+
+        # Get parameters
+        self.computer_or_policy=params.get(Input.COMPUTER_OR_POLICY)
+        self.id=params.get(Input.ID)
+        self.rules=params.get(Input.RULES)
+
+        # Check if the rules should be assigned to a computer or policy
+        if self.computer_or_policy == "computer":
+            url = f"{self.connection.dsm_url}/api/computers/{self.id}/intrusionprevention/assignments"
+        else:
+            url = f"{self.connection.dsm_url}/api/policies/{self.id}/intrusionprevention/assignments"
+
+        self.logger.info("Setting rules: ")
+        self.logger.info(self.rules)
+
+        # Prepare request
+        data = { "ruleIDs": self.rules }
+
+        post_header = {
+                        "Content-type": "application/json",
+                        "api-secret-key": self.connection.dsm_api_key,
+                        "api-version": "v1"
+                        }
+
+        # Set rules
+        response = requests.post(url,
+                                data=json.dumps(data),
+                                headers=post_header,
+                                verify=True)
+        response.close()
```

Comment:
This is not needed unless you found it to be necessary during your testing.

Reply:
done